src/prefect/task_worker.py

```diff
@@ -76,28 +84,30 @@ class TaskWorker:
 
     def __init__(
         self,
-        *tasks: Task,
+        *tasks: Task[P, R],
         limit: Optional[int] = 10,
     ):
-        self.tasks = []
+        self.tasks: list[Task[Any, Any]] = []
         for t in tasks:
-            if isinstance(t, Task):
+            if isinstance(t, Task):  # pyright: ignore[reportUnnecessaryIsInstance]
```

Comment:
I prefer to hide these checks in a `if not TYPE_CHECKING:` block. After all, only users that don't use type checking will need this runtime check ;-)